### PR TITLE
refactor: governance constification and deglobalization

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Some relatively simple refactoring; inspired by reviewing #5569; adds some constification and some deglobalization

## What was done?
Partial deglobalization and constification

## How Has This Been Tested?
Building

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/masternode/sync.cpp [hunk 1]
```diff
@@ -16,10 +16,11 @@
 class CMasternodeSync;
 std::unique_ptr<CMasternodeSync> masternodeSync;
 
-CMasternodeSync::CMasternodeSync(CConnman& _connman) :
```
Comment (kwvg) [inline src/masternode/sync.cpp:19]: Consider naming it `govman`, similar to [`dsnotificationinterface.cpp`](https://github.com/dashpay/dash/blob/b20ff5406cf5a6673babed07578c3e4b542d1456/src/dsnotificationinterface.h#L45). 

It's very unlikely `govman` would be confused with something else as no manager shares its name or has a similar one.

File: src/net_processing.cpp [hunk 2]
```diff
@@ -1542,21 +1543,22 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex, const Conse
...
-                                 const std::unique_ptr<LLMQContext>& llmq_ctx, bool ignore_incoming_txs)
+                                 const std::unique_ptr<LLMQContext>& llmq_ctx, CGovernanceManager& gov_man, bool ignore_incoming_txs)
     : m_chainparams(chainparams),
       m_connman(connman),
       m_addrman(addrman),
       m_banman(banman),
       m_chainman(chainman),
       m_mempool(pool),
       m_llmq_ctx(llmq_ctx),
+      m_governance_manager(gov_man),
```
Comment (kwvg) [inline src/net_processing.cpp:1561]: Same here, long names make for unwieldy typing.

**Review by kwvg — COMMENTED** [review]
Concept ACK, minor nits.

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by kwvg — APPROVED** [review]
utACK

**Pull request merged**